staging/src/k8s.io/client-go/transport/ca_rotation.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transport
+
+import (
+	"bytes"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+	"k8s.io/utils/clock"
+)
+
+var caRefreshDuration = 5 * time.Minute
+
+// atomicTransportHolder holds a transport that can be atomically updated
+// when CA files change, enabling graceful CA rotation without cache complexity
+type atomicTransportHolder struct {
+	caFile        string
+	config        *Config
+	currentCAData []byte // Track the actual CA data currently in use
+	// clock is used to allow for testing time-based logic.
+	clock clock.Clock
+	// mu covers transport and transportLastUpdated
+	mu                   sync.RWMutex
+	transport            *http.Transport
+	transportLastChecked time.Time
+}
+
+// RoundTrip implements http.RoundTripper interface
+func (h *atomicTransportHolder) RoundTrip(req *http.Request) (*http.Response, error) {
+	transport, err := h.getTransport()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get fresh transport: %w", err)
```

Comment:
I don't think we *ever* want a CA reload failure that was just a periodic retry to turn into a request failure when we previously had a usable transport

Normally when a dynamic reload fails, we turn that into a metric or something, but keep using the previously successfully loaded data